examples/ec2/main.tf

```diff
@@ -0,0 +1,201 @@
+    EOF
+  EOT
+
+  tags = {
+    Name       = local.name
+    Example    = local.name
+    Repository = "https://github.com/terraform-aws-modules/terraform-aws-ecs"
+  }
+}
+
+################################################################################
+# ECS Module
+################################################################################
+
+module "ecs_disabled" {
+  source = "../.."
+
+  create = false
+}
+
+module "ecs" {
+  source = "../.."
+
+  cluster_name = local.name
+
+  cluster_configuration = {
+    execute_command_configuration = {
+      logging = "OVERRIDE"
+      log_configuration = {
+        # You can set a simple string and ECS will create the CloudWatch log group for you
+        # or you can create the resource yourself as shown here to better manage retetion, tagging, etc.
+        # Embedding it into the module is not trivial and therefore it is externalized
+        cloud_watch_log_group_name = aws_cloudwatch_log_group.this.name
+      }
+    }
+  }
+
+  # Capacity provider - autoscaling groups
+  autoscaling_capacity_providers = {
+    one = {
+      auto_scaling_group_arn         = module.autoscaling["one"].autoscaling_group_arn
+      managed_termination_protection = "ENABLED"
+
+      managed_scaling = {
+        maximum_scaling_step_size = 5
+        minimum_scaling_step_size = 1
+        status                    = "ENABLED"
+        target_capacity           = 60
+      }
+
+      default_capacity_provider_strategy = {
+        weight = 60
+        base   = 20
+      }
+    }
+    two = {
+      auto_scaling_group_arn         = module.autoscaling["two"].autoscaling_group_arn
+      managed_termination_protection = "ENABLED"
+
+      managed_scaling = {
+        maximum_scaling_step_size = 15
+        minimum_scaling_step_size = 5
+        status                    = "ENABLED"
+        target_capacity           = 90
+      }
+
+      default_capacity_provider_strategy = {
+        weight = 40
+      }
+    }
+  }
+
+  tags = local.tags
+}
+
+module "hello_world" {
+  source = "./service-hello-world"
+
+  cluster_id = module.ecs.cluster_id
+}
+
+################################################################################
+# Supporting Resources
+################################################################################
+
+# https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-optimized_AMI.html#ecs-optimized-ami-linux
+data "aws_ssm_parameter" "ecs_optimised_ami" {
+  name = "/aws/service/ecs/optimized-ami/amazon-linux-2/recommended"
+}
+
+module "autoscaling" {
+  source  = "terraform-aws-modules/autoscaling/aws"
+  version = "~> 6.5"
+
+  for_each = {
+    one = {
+      instance_type = "t3.micro"
+    }
+    two = {
+      instance_type = "t3.small"
+    }
+  }
+
+  name = "${local.name}-${each.key}"
+
+  image_id          = jsondecode(data.aws_ssm_parameter.ecs_optimised_ami.value)["image_id"]
+  instance_type     = each.value.instance_type
+  ebs_optimized     = true
```

Comment:
Cheaper options (e.g. without EBS optimized) are better for examples, and sometimes quicker to spin up.